docker/docker_image_src.go

```diff
@@ -30,15 +32,68 @@ type dockerImageSource struct {
 
 // newImageSource creates a new ImageSource for the specified image reference.
 // The caller must call .Close() on the returned ImageSource.
```

Comment:
Can we extend the comment to reflect the new semantics? That may be redundant to many other places but it's a fairly important part worth mentioning.

Reply:
I updated the documentation which should now represent the new behavior of the function.